### ISSUE TITLE
Add guardrail tracing filters to prevent logging of sensitive plaintext 

### ISSUE THREAD

**Issue by divergentdave** [divviup/janus#2578]
Some of our dependencies may call `log` or `tracing` macros with raw contents of HTTP bodies or SQL query parameters at sufficiently high logging levels. We could identify these log messages, and then add filters for the specific targets to set their level to OFF, overriding broader filters with higher level from environment variables.

Examples:

> DEBUG tokio_postgres::query: executing statement s263 with parameters: [...]

> TRACE trillium_http::conn: received:
POST /tasks/.../aggregate_shares HTTP/1.1
Accept: */*
Content-Type: application/dap-aggregate-share-req
User-Agent: janus_aggregator/0.7.0-prerelease-1/collection_job_driver
Content-Length: 61
Host: 127.0.0.1:44447
dap-auth-token: ...

We could probably test this, but it would be a bit involved. We can create a tracing subscriber that we can read from for validation, but then our choices are to either install the subscriber globally, which implies a separate integration test target, or to pass a closure to `tracing::subscriber::with_default()`, which we would have to build into a custom async runtime, perhaps wrapping Tokio's multi-threaded runtime.

**Comment by jbr** [comment]
In the case of trillium I'd also be open to supporting this as a feature; it's a common enough concern

**Comment by jbr** [comment]
I don't believe trillium ever logs bodies, but trillium-http could have a set of HeaderNames to scrub the values of on logging

**Comment by jbr** [comment]
Alternatively I am not strongly convinced header tracing needs to be a trillium-http concern
